### ISSUE TITLE
Running DESeq2 with more then two conditions.

### ISSUE THREAD

**Issue by tijsbliek** [KoesGroup/Snakemake_hisat-DESeq#2]
As it is now, the the pipeline (in the branch develop) works but can only compare two different conditions. So It can get differetial exrpessions between a wt and a mutant or control and treated sample but not between a wt and multiple mutants or maybe wt and mutant in different tissues.

I would like this to be implemented.

For that I have added a sample2.tsv in the folder data. In this sample file there are 4 different genotypes and two different tissues. Making a total of seven (one combination is missing) different conditions. Also I have added the raw counts in the file, results/counts2.txt.

I guess the DESeq2.R should be changed so it can read the conditions from the file (samples2.tsv) and no longer from the command line as it is doing now.  And somehow a "contrast" should be added in the R script but so far I have been unsuccesfull...  I am a bit stuck would like some help on this subject. If there is an other way without the use of contrast I would be very happy too.

yours,
tijs

**Comment by mgalland** [comment]
Hey @tijsbliek. Could you attach a `counts` and a `samples` files to that issue?

**Comment by mgalland** [comment]
Yes and no! I meant attaching files directly to this issue threads as it would make files easier to find. For instance, I have uploaded the DESeq2 manual in this comment.
[DESeq2.pdf](https://github.com/KoesGroup/Snakemake_hisat-DESeq/files/2599628/DESeq2.pdf)

**Comment by tijsbliek** [comment]
I missed the fact that files can be dropped in this box. I will remember for a next time.

Also with the contrast it seems like it is only possible to only compare two conditions. Or do you think I should iterate through the list and do all combinations???

**Comment by mgalland** [comment]
I have uploaded my changes on the DESeq2 script so far. I try to deal with multiple conditions. Can you have a look at it?
https://github.com/KoesGroup/Snakemake_hisat-DESeq/blob/deseq/scripts/DESeq2.R

**Comment by tijsbliek** [comment]
Thanks,

It looks pretty complicated, but I'll have a look.

**Comment by tijsbliek** [comment]
optparse is in quotation marks, shouldn't be should it?

library("optparse")

**Comment by tijsbliek** [comment]
ok. good to know.

**Comment by tijsbliek** [comment]
The reading of the counts data should have this "skip =1". It has the first line containing the command for the featureCounts.
countdata <- read.table(opt$counts, header=TRUE, skip=1, row.names=1,stringsAsFactors = F)

**Comment by tijsbliek** [comment]
I wrote the R-script and it works on my desktop R-studio but when I want to implement it to the geneses pipeline I get this error:
Error in parse_args(opt_parser) : 
  Error in getopt(spec = spec, opt = args) : 
  "counts2.txt" is not a valid option, or does not support an argument
Execution halted

any clue???

**Comment by tijsbliek** [comment]
I solved the problem with the arguments

Also I have added the DESeq2.R to the branch deseq, overwriting the one you wrote.
It still iterates through the list to get all the posible combinations of conditions, and also I added an extra feature where it can exclude condition combinations that are to different. To be chosen as a command line argument (-m/--maxfraction). Advantage of this is that like in the case of petunia, if the maxfraction is set to 0.1 it will include all the leaf-leaf and petal-petal combinations but not leaf-petal combinations as they are just to different(more then 10% in this case). 
So it is most likely not as readable as you would probably like it to be (eventhough I have added a lot of commands) but I am quite happy with the way it works.
